pydicom/dataelem.py

```diff
@@ -302,10 +302,17 @@ def __str__(self):
     @property
     def repval(self):
         """Return a str representation of the element's `value`."""
-        byte_VRs = ['OB', 'OW', 'OW/OB', 'OW or OB', 'OB or OW',
-                    'US or SS or OW', 'US or SS']
-        if (self.VR in byte_VRs and len(self.value) > self.maxBytesToDisplay):
-            repVal = "Array of %d bytes" % len(self.value)
+        long_VRs = {"OB", "OD", "OF", "OW", "UN", "UT"}
+        if set(self.VR.split(" or ")) & long_VRs:
+            try:
+                length = len(self.value)
+            except TypeError:
+                pass
+            else:
+                if length > self.maxBytesToDisplay:
+                    return "Array of %d elements" % length
+        if self.VM > self.maxBytesToDisplay:
```

Comment:
Shouldn't you check the length of all values here? This check only works as expected for OB, I think

Reply:
This doesn't actually check the length in bytes, rather in elements, that's true. Perhaps we could change the name of this attribute? It seems to me to be wasteful, to try and determine one size are our elements (1 byte for OB, 2 for OW, etc) only for truncating the display.
Either way, I'm quite sure this checks all cases - the variable sized VRs are the one in the set and are tested for length; anything else can only be long if it has a large VM.

Reply:
> anything else can only be long if it has a large VM

Right - I mean the representation of these could be of quite different length if it depends only on the VM. There can be large VMs for different VRs, like DS (as in RTStruct, I think), which would take much more space than an OB of the same length. But I agree that we probably don't want to put too much of effort here - the user will always be able to restrict the output by simple formatting options.

Reply:
> Perhaps we could change the name of this attribute?

How about a new attribute - e.g. self.max_VM_to_display (now in PEP8 style)?  I think that would allow separation of display truncation from ridiculously long outputs, as intended by this change.